api/controllers/MerchStoreController.ts

```diff
@@ -167,16 +170,22 @@ export class MerchStoreController {
   @Post('/order')
   async placeMerchOrder(@Body() placeOrderRequest: PlaceMerchOrderRequest,
     @AuthenticatedUser() user: UserModel): Promise<PlaceMerchOrderResponse> {
-    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
-    const originalOrder = placeOrderRequest.order.filter((oi) => oi.quantity > 0);
-    const orderIsEmpty = originalOrder.reduce((x, n) => x + n.quantity, 0) === 0;
-    if (orderIsEmpty) throw new UserError('There are no items in this order');
-    const numUniqueUuids = (new Set(originalOrder.map((oi) => oi.option))).size;
-    if (originalOrder.length !== numUniqueUuids) throw new BadRequestError('There are duplicate items in this order');
+    const originalOrder = this.verifyMerchOrderRequest(placeOrderRequest.order, user);
+
     const order = await this.merchStoreService.placeOrder(originalOrder, user);
     return { error: null, order };
   }
 
+  @Post('/order/verification')
+  async verifyMerchOrder(@Body() verifyOrderRequest: VerifyMerchOrderRequest,
+    @AuthenticatedUser() user: UserModel): Promise<VerifyMerchOrderResponse> {
+    const originalOrder = this.verifyMerchOrderRequest(verifyOrderRequest.order, user);
+
+    await this.merchStoreService.verifyOrder(originalOrder, user);
+
+    return { error: null };
+  }
```

Comment:
nit: I don't think the extra lines are needed, we can keep the same convention that the other routes have with spacing.